docs/design/node-selection.md

```diff
@@ -0,0 +1,82 @@
+
+> Total Repair Reputation = uptime_repair_weight · uptime R(n) + audit_repair_weight · audit R(n)
+>
+> Total Uplink Reputation = uptime_uplink_weight · uptime R(n) + audit_uplink_weight · audit R(n)
+
+This design may be refined in the future to prefer faster nodes, geography, etc..
+
+### Database changes
+
+```DBX
+model node (
+...
+	field audit_reputation_a   float64   ( updatable )
+	field audit_reputation_b   float64   ( updatable )
+	field total_audit_count    int64     ( updatable )
+	field uptime_reputation_a  float64   ( updatable )
+	field uptime_reputation_b  float64   ( updatable )
+	field total_uptime_count   int64     ( updatable )
+...
+)
+```
+
+## Rationale
+
+The Storj Data Science team has currently published two papers on the design of our reputation score:
+[Reputation Scoring](https://github.com/storj/datascience/blob/master/reputation/Reputation_Scoring_Framework_Highlevel.pdf and [Extending Audit/Uptime Success Ratios](
+https://github.com/storj/datascience/blob/master/reputation/extending%20ratios%20to%20reputation/extending%20ratios%20to%20reputation.pdf).  
+
+These papers put forth a model where reputation chance be determined based on previous reputation values α and β, a forgetting factor λ, single value feedback _v_, and a normalization weight _w_.
+
+> We suggest one minor modification from [1] to how _v_ is selected. ... we propose starting with small (but negative) values for failures, and small (but positive) values for successes.... This has the benefit of "easing" a new node's reputation towards its "true" reputation. ... We start with α0 = 1 and β0 = 1 for two reasons:  first, because 0/0 is undefined; second, this assigns new nodes a reputation score of 0.5...
+
+Implementing α0 = β0 = 1 as described above would require some type of relaxation of disqualification criteria for new nodes.  The alternative recommended in this document is to initialize α0 and β0 such that α0 / (α0 + β0) is greater than the disqualification cutoff and less than or equal to 1 (the maximum reputation).  The easing of _v_ described above may lessen in importance, given these starting values.  For the initial implementation, we assume that _v_ is fixed at 1.
+
+## Implementation
+
+* Create configuration elements for audit_a0, audit_β0, audit_λ, audit_w, audit_repair_weight, and audit_uplink_weight
+* Create configuration elements for uptime_a0, uptime_β0, uptime_λ, uptime_w, uptime_repair_weight, and uptime_uplink_weight
+* Alter DBX model removing audit_success_count, audit_success_ratio, uptime_success_count, uptime_ratio
+* Alter DBX model adding audit_reputation_a, audit_reputation_b, uptime_reputation_a, uptime_reputation_b
+* Alter SQL node selection queries to consider new values
+* Alter SQL node selection queries to return 2x more nodes
+* Implement "Power of Two Choices" logic in node selection query
+* Update disqualification code to use reputation
```

Comment:
I'm not entirely sure about this task.
Is the disqualification implemented? my main understanding is that that disqualification is mostly a flag to mark a node when it is disqualified, currently there is not such logic, there is only the database column to mark the node and the node selection queries has been updated to ignore those nodes which are marked as disqualified.

With the creation of this document, I see that the [Node Disqualification document](https://github.com/storj/storj/blob/master/docs/design/disqualification.md) could have some modifications/adaptations for example in the section [Determining Disqualification](https://github.com/storj/storj/blob/master/docs/design/disqualification.md#determining-disqualification) could be simplified that it's based in the reputation and link to this document.

Reply:
Better?

Reply:
Yes, it's clearer now.

However I'm still thinking that the task should be in the _Node Disqualification document_ because as far as I know it isn't already implemented, right?

Reply:
I don't have a strong opinion on where this work lands.  It can be one PR or two, I assumed it'd be two PRs.

Reply:
Well, then it must exist in order to be updated, but if it doesn't exist I'm not sure what we'll have to update.
Anyway, we can figure it out at the time to work on it.